tests/components/demo/test_notify.py

```diff
@@ -8,189 +9,198 @@
-        step = {
-            "service": "notify.notify",
-            "data": {
-                "data": {
-                    "push": {"sound": "US-EN-Morgan-Freeman-Roommate-Is-Arriving.wav"}
-                }
-            },
-            "data_template": {"message": "Test 123 {{ 2 + 2 }}\n", "title": "Test"},
-        }
-        setup_component(self.hass, "script", {"script": {"test": {"sequence": step}}})
-        self.hass.services.call("script", "test")
-        self.hass.block_till_done()
-        assert len(self.events) == 1
-        assert {
-            "message": "Test 123 4",
-            "title": "Test",
-            "data": {
-                "push": {"sound": "US-EN-Morgan-Freeman-Roommate-Is-Arriving.wav"}
-            },
-        } == self.events[0].data
-
-    def test_targets_are_services(self):
-        """Test that all targets are exposed as individual services."""
-        self._setup_notify()
-        assert self.hass.services.has_service("notify", "demo") is not None
-        service = "demo_test_target_name"
-        assert self.hass.services.has_service("notify", service) is not None
-
-    def test_messages_to_targets_route(self):
-        """Test message routing to specific target services."""
-        self._setup_notify()
-        self.hass.bus.listen_once("notify", self.record_calls)
-
-        self.hass.services.call(
-            "notify",
-            "demo_test_target_name",
-            {"message": "my message", "title": "my title", "data": {"hello": "world"}},
-        )
 
-        self.hass.block_till_done()
+@pytest.fixture
+def calls():
+    """Fixture to calls."""
+    return []
+
 
-        data = self.calls[0][0].data
+@pytest.fixture
+def record_calls(calls):
+    """Fixture to record calls."""
 
-        assert {
-            "message": "my message",
-            "target": ["test target id"],
-            "title": "my title",
-            "data": {"hello": "world"},
-        } == data
+    @callback
+    def record_calls(*args):
+        """Record calls."""
+        calls.append(args)
+
+    return record_calls
+
+
+@pytest.fixture(name="mock_demo_notify")
+def mock_demo_notify_fixture():
+    """Mock demo notify service."""
+    with patch("homeassistant.components.demo.notify.get_service", autospec=True) as ns:
+        yield ns
+
+
+async def setup_notify(hass):
+    """Test setup."""
+    with assert_setup_component(1, notify.DOMAIN) as config:
+        assert await async_setup_component(hass, notify.DOMAIN, CONFIG)
+    assert config[notify.DOMAIN]
+    await hass.async_block_till_done()
+
+
+async def test_no_notify_service(hass, mock_demo_notify, caplog):
+    """Test missing platform notify service instance."""
+    caplog.set_level(logging.ERROR)
+    mock_demo_notify.return_value = None
+    await setup_notify(hass)
+    await hass.async_block_till_done()
+    assert mock_demo_notify.called
+    assert "Failed to initialize notification service demo" in caplog.text
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)